### PR TITLE
MODQM-78 Implement GET status endpoint

### PR DESCRIPTION
**Pull request by psmagin**
## Purpose
Support endpoint to get status of MARC bibliographic record creation

## Approach
* add new endpoint to API declaration
* create liqb scripts to create `record_creation_status` table
* implement endpoint
* update API tests to work with database

### CODE REVIEW

File: src/main/resources/swagger.api/schemas/creationStatus.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          "description": "Date and time of creation of the status",
+          "type": "string",
+          "format": "date-time",
+          "example": "2020-07-14T12:20:10.000"
+        },
+        "updatedAt": {
+          "description": "Date and time of last successful update of the status",
+          "type": "string",
+          "format": "date-time",
+          "example": "2020-07-14T12:20:10.000"
```
Comment (dmtkachenko) [inline src/main/resources/swagger.api/schemas/creationStatus.json:49]: what about user who updated/created

File: src/test/java/org/folio/qm/controller/BaseApiTest.java [hunk 2]
```diff
@@ -1,73 +1,114 @@
...
       .extract()
       .response();
   }
+
+  private String getRequestUrl(String path) {
+    return "http://localhost:" + port + path;
+  }
+
+
+  private String creationStatusTable() {
```
Comment (dmtkachenko) [inline src/test/java/org/folio/qm/controller/BaseApiTest.java:105]: Why are these status related methods present in the base class?

File: src/test/java/org/folio/qm/extension/DatabaseInitializer.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.context.event.ContextClosedEvent;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+public class DatabaseInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+  @Override
+  public void initialize(ConfigurableApplicationContext configurableApplicationContext) {
+    JdbcDatabaseContainer<?> container = new PostgreSQLContainer<>("postgres:12-alpine").waitingFor(Wait.forListeningPort());
```
Comment (dmtkachenko) [inline src/test/java/org/folio/qm/extension/DatabaseInitializer.java:15]: testcontainers are good but only if there is docker available in the workstation running the tests
consider using https://github.com/zonkyio/embedded-database-spring-test

File: pom.xml [hunk 4]
```diff
@@ -203,11 +233,14 @@
...
               <generateApiDocumentation>true</generateApiDocumentation>
               <generateModels>true</generateModels>
-              <modelsToGenerate>error,errors,parameter,quickMarc,quickMarc_fields,quickMarc_updateInfo</modelsToGenerate>
+              <modelsToGenerate>creationStatus,creationStatus_metadata,quickMarc,quickMarc_fields,quickMarc_updateInfo</modelsToGenerate>
               <generateModelTests>false</generateModelTests>
               <generateSupportingFiles>true</generateSupportingFiles>
               <supportingFilesToGenerate>ApiUtil.java</supportingFilesToGenerate>
               <generateModelDocumentation>true</generateModelDocumentation>
+              <importMappings>
+                <importMapping>error=org.folio.tenant.domain.dto.Error</importMapping>
```
Comment (dmtkachenko) [inline pom.xml:242]: (like)

File: src/main/resources/swagger.api/schemas/creationStatus.json (hunk 1 above)
Comment (psmagin) [reply src/main/resources/swagger.api/schemas/creationStatus.json:49]: I'm not sure that we need that information.
The status update will be performed by an event consumer only and there is no user. 
A user that created - maybe but also not sure

**Review by dmtkachenko — APPROVED** [review]

**Review by natali-zaitseva — APPROVED** [review]

**Comment by dcrossleyau** [comment]
@psmagin and @natali-zaitseva : The basic raml error reported above, and merged without fixing, will occur on mainline builds and on subsequent pull-requests.

**Comment by dcrossleyau** [comment]
As noted in the api-lint messages above, the error is with the RAML file processing, not the OpenAPI YAML file. The RAML file has an extra colon on line 79.
